### PR TITLE
fix(Client): Fix UI

### PR DESCRIPTION
**Pull request by BrianBts**

### CODE REVIEW

File: src/common/StatusColors.tsx [hunk 1]
```diff
@@ -35,4 +35,33 @@ export const StateColors: Record<OrderStatus, StatusColor> = {
   PREPARING: "warn",
   AVAILABLE: "pending",
   SHIPPED: "success",
-} as const;
\ No newline at end of file
+} as const;
+
+export const JobPositionColors: Record<string, string> = {
```
Comment (sebikap-gm2) [inline src/common/StatusColors.tsx:40]: Just do a list of 20 colors and go one job position by one taking one color

File: src/components/TableDetails.tsx [hunk 2]
```diff
@@ -56,9 +57,9 @@ export const TableDetails = observer(function ({
                 {shipment.name} {shipment.lastName}
               </b>
             </td>
-            <td className="  py-4 px-3">
-              <State message={shipment.status} className="p-1" />
-            </td>
+            <td className="py-4 px-3">
+            <State status={mapStatusToOrderStatus(shipment.status)} className="p-1" />
```
Comment (sebikap-gm2) [inline src/components/TableDetails.tsx:61]: `text-transform: uppercase;`

File: src/components/Table.tsx [hunk 3]
```diff
@@ -14,64 +14,82 @@ type TableProps = {
...
+    }, {});
+    setRowOpenState(newState);
+  }, [products]);
 
-  //TODO:  to toggle by rowId instead of index
-  const toggleRow = (index: number) => {
-    const updatedRowOpenState = [...rowOpenState];
-    updatedRowOpenState[index] = !updatedRowOpenState[index];
-    setRowOpenState(updatedRowOpenState);
+  const toggleRow = (productId: any) => {
```
Comment (sebikap-gm2) [inline src/components/Table.tsx:30]:
```
any -> string
any -> Product["_id"]
```

File: src/app/home/my-team/page.tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ export default function MyTeam() {
             variant="primary"
             size="big"
             className="rounded-md flex gap-2"
-            href="#"
+            href=""
```
Comment (sebikap-gm2) [inline src/app/home/my-team/page.tsx:35]: Redirect to add team member

File: src/common/Graphs.tsx [hunk 5]
```diff
@@ -8,21 +8,25 @@ interface DoughnutChartProps {
...
+    stock?: number
   };
 }
 
 export function DoughnutChart({ className, data }: DoughnutChartProps) {
-  const number = 10;
-  const quantity = data && data.quantity ? data.quantity : 0;
+  const stock = data.stock || 0;
+  //TODO: check this, where are assigned and available values?
+  const quantity = data.quantity || 20;
```
Comment (sebikap-gm2) [inline src/common/Graphs.tsx:18]: Do not default this value

**Review by sebikap-gm2 — CHANGES_REQUESTED** [review]

File: src/common/Layout.tsx [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
   children: ReactNode;
   className?: string;
-};
+}
 
 export function Layout({ children, className }: LayoutProps) {
   return (
     <section
-      className={`h-[calc(98vh-100px)] w-[96%] mx-auto py-6 mt-6 ${className}`}
+      className={`h-auto min-h-full mx-auto    flex-grow   ${className}`}
```
Comment (sebikap-gm2) [inline src/common/Layout.tsx:11]: Please check space in between class names or I will die

File: src/common/Graphs.tsx [hunk 7]
```diff
@@ -57,20 +61,26 @@ export function DoughnutChart({ className, data }: DoughnutChartProps) {
...
-        <div className="flex gap-1 items-center">
-          <div className="h-[1rem] w-[1rem] bg-purple rounded-full"></div>
+        <div className="flex gap-1 items-center pt-8">
+          <div
+            className="h-[1rem] w-[1rem]"
+            style={{ backgroundColor: assignedColor }}
+          ></div>
           <p>
-            Assigned | <b> {quantity} </b>
+            Assigned | <b> {quantity + 20} </b>
```
Comment (sebikap-gm2) [inline src/common/Graphs.tsx:74]: Remove horrendous hardcode

File: src/common/Graphs.tsx [hunk 8]
```diff
@@ -8,26 +8,30 @@ interface DoughnutChartProps {
...
-  const number = 10;
-  const quantity = data && data.quantity ? data.quantity : 0;
+  const stock = data.stock || 0;
+  //TODO: check this, where are assigned and available values?
+  const quantity = data.quantity;
+
+  const assignedColor = "#9747FF";
+  const availableColor = "#4FE8B7";
+
   const info = {
```
Comment (sebikap-gm2) [inline src/common/Graphs.tsx:23]: Check if we can type this

File: src/common/EquipmentRow.tsx [hunk 9]
```diff
@@ -36,7 +36,7 @@ export function EquipmentRow({
         {order._id}
       </td>
       <td className="pl-3 py-3">
-        {selectedMember.firstName} {selectedMember.lastName}
+        {selectedMember?.firstName} {selectedMember?.lastName}
```
Comment (sebikap-gm2) [inline src/common/EquipmentRow.tsx:39]: Remove component

File: src/app/home/orders/page.tsx [hunk 10]
```diff
@@ -1,17 +1,20 @@
...
-export default observer( function OrderPage() {
-  const {orders:{orders}} =  useStore()
+export default observer(function OrderPage() {
+  const {
+    orders: { orders },
+  } = useStore();
   return (
     <div>
-    {
+      {/* {
```
Comment (sebikap-gm2) [inline src/app/home/orders/page.tsx:14]: Review

File: src/app/home/my-team/DataTeam.tsx [hunk 11]
```diff
@@ -13,78 +13,79 @@ import { observer } from "mobx-react-lite";
...
+  return (
+    <div className="flex flex-col gap-4 w-full h-full p-4">
+      <div className="w-full flex  justify-end gap-2 ">
+        <CustomLink
+          className={"rounded-md text-sm p-2 flex items-center gap-2"}
+          variant={"secondary"}
+          href="/home/addTeam"
+        >
+          <AddIcon className={"h-4 w-4"} /> Add Team Member
+        </CustomLink>
```
Comment (sebikap-gm2) [inline src/app/home/my-team/DataTeam.tsx:44]: Move to own component (Maybe MyTeamAppBar/MyTeamHeader)

File: src/app/home/my-stock/EmptyStock.tsx [hunk 12]
```diff
@@ -2,40 +2,41 @@
...
+  } = useStore();
+
+  return (
+    <div className="border-2 shadow-sm border-border rounded-md grid place-items-center w-[98%] ">
+      <EmptyCard
+        imageBottom="/office.svg"
+        paragraph="You don't have any items."
+        altImage="Office"
+      >
+        <div className="flex gap-2 ">
```
Comment (sebikap-gm2) [inline src/app/home/my-stock/EmptyStock.tsx:18]: Move Empty state component to own component and use everywhere

File: src/app/home/layout.tsx [hunk 13]
```diff
@@ -46,15 +39,15 @@ export default observer(function RootLayout({ children }: RootLayoutProps) {
         });
       }
     }
-  }, [session]);
+  }, [session, store.members]);
```
Comment (sebikap-gm2) [inline src/app/home/layout.tsx:42]:
```
const {
   members: { setMembers }
} = useStore();
```

Add to existing line 16

**Review by sebikap-gm2 — CHANGES_REQUESTED** [review]

File: src/common/StatusColors.tsx (hunk 1 above)
Comment (Francisco-Villanueva) [reply src/common/StatusColors.tsx:40]: Do you mean something like that?

`export const JobPositionColors = [
  "bg-pink-400",
  "bg-green",
  "bg-purple",
  "bg-yellow-200",
  "bg-red-200",
];`

File: src/common/Graphs.tsx (hunk 8 above)
Comment (Francisco-Villanueva) [reply src/common/Graphs.tsx:23]: Fixed on ['fix/graph-stock-view'](https://github.com/sebikap-gm2/First-Plug-Platform-Client/pull/135) PR

File: src/components/TeamMembers.tsx [hunk 14]
```diff
@@ -9,10 +9,9 @@ interface TeamMembersProps {
 }
 
 export const TeamMembers = observer(function ({ display }: TeamMembersProps) {
-
   return (
-    <section className="flex flex-col gap-4">
-      {display === "grid" ? <GridTeam  /> : <TableTeam />}
+    <section className="flex flex-col gap-4 absolute  bottom-0 left-0 overflow-auto  h-[80%] w-full ">
```
Comment (sebikap-gm2) [inline src/components/TeamMembers.tsx:13]: We should avoid absolute positioning, it should be used only in certain cases. What was the issue here? There probably is a better solution using flexbox

File: src/components/TeamHomeCard.tsx [hunk 15]
```diff
@@ -1,50 +1,49 @@
...
 import { useStore } from "@/models";
 import { MyTeamHeader } from "./MyTeamHeader";
 
-interface TeamCardProps {
-  className?: string;
-}
+interface TeamCardProps {}
 
-export const TeamHomeCard = observer(function ({ className }: TeamCardProps) {
+export const TeamHomeCard = observer(function ({}: TeamCardProps) {
```
Comment (sebikap-gm2) [inline src/components/TeamHomeCard.tsx:10]: Avoid. No props -> no props object

File: src/components/Card.tsx [hunk 16]
```diff
@@ -23,17 +23,15 @@ export const Card = function ({
...
-  className,
+  className = "",
   onClick,
 }: CardProps) {
   return (
     <article
-      className={`bg-white rounded-[16px] p-6 border border-grey ${
-        className || ""
-      }`}
+      className={` felx flex-col  rounded-xl p-4  border border-border  relative ${className}`}
```
Comment (sebikap-gm2) [inline src/components/Card.tsx:31]: felx?

File: src/components/Card.tsx [hunk 17]
```diff
@@ -49,18 +47,27 @@ export const Card = function ({
...
-        }  gap-y-[.5rem] `}
-      >
-        {imageBottom && (
-          <div className="w-52 h-52 relative">
-            <Image src={imageBottom} alt={altImage} fill priority />
+      <div>
+        {children ? (
+          <div
+            className={` ${
+              Title ? "" : "absolute"
```
Comment (sebikap-gm2) [inline src/components/Card.tsx:54]: Avoid absolute. What's the issue here?

File: src/common/EmptyCard.tsx [hunk 18]
```diff
@@ -1,21 +1,85 @@
...
+
+interface EmptyCardProps {
   className?: string;
   imageBottom: string;
   altImage: string;
   paragraph: string;
-  children: ReactNode
+  children: childrenTypes;
 }
+function EmptyCardChildren({ children }: { children: childrenTypes }) {
```
Comment (sebikap-gm2) [inline src/common/EmptyCard.tsx:16]: This is a step forward, but no exactly what I had in mind. Let's review together @Francisco-Villanueva

File: src/common/DropFilter.tsx [hunk 19]
```diff
@@ -1,29 +1,31 @@
...
 
 interface DropFilterProps {
   children?: ReactNode;
-  body?: string;
-  className?: string;
+  contentText?: string;
 }
 
-export function DropFilter({ children, body, className }: DropFilterProps) {
+export function DropFilter({ children, contentText }: DropFilterProps) {
```
Comment (sebikap-gm2) [inline src/common/DropFilter.tsx:10]: Let's review together this component type

File: src/app/home/shipments/EmptyShipments.tsx [hunk 20]
```diff
@@ -1,22 +1,14 @@
...
-        size="big"
-        className="rounded-md flex justify-center"
-        href="/home/my-stock"
-      >
-        Got to My Stock
-      </CustomLink>
-    </EmptyCard>
-  </Layout>
-  )
+    <EmptyCardLayout>
```
Comment (sebikap-gm2) [inline src/app/home/shipments/EmptyShipments.tsx:5]: Review all together with Empty Card things

**Review by sebikap-gm2 — CHANGES_REQUESTED** [review]

**Pull request merged**